2008-09-16  Ulrich Drepper  <drepper@redhat.com>

	* dlfcn/dlopen.c (dlopen_doit): Allow __RTLD_SPROF flag.
	Patch by Olivier Fourdan <ofourdan@redhat.com>.

--- libc/dlfcn/dlopen.c	13 Jan 2005 02:03:26 -0000	1.12
+++ libc/dlfcn/dlopen.c	17 Sep 2008 03:12:24 -0000	1.13
@@ -60,7 +60,8 @@ dlopen_doit (void *a)
   struct dlopen_args *args = (struct dlopen_args *) a;
 
   if (args->mode & ~(RTLD_BINDING_MASK | RTLD_NOLOAD | RTLD_DEEPBIND
-		     | RTLD_GLOBAL | RTLD_LOCAL | RTLD_NODELETE))
+		     | RTLD_GLOBAL | RTLD_LOCAL | RTLD_NODELETE
+		     | __RTLD_SPROF))
     GLRO(dl_signal_error) (0, NULL, NULL, _("invalid mode parameter"));
 
   args->new = GLRO(dl_open) (args->file ?: "", args->mode | __RTLD_DLOPEN,
